v8env/src/bridge.ts

```diff
@@ -105,29 +106,38 @@ export function addEventListener(name: string, fn: Function) {
             }) : null
         })
 
-        try {
-          fn.call(window, {
-            request: req,
-            respondWith(resfn: any) {
-              try {
-                let ret = resfn;
-                if (typeof ret === "function") {
-                  ret = resfn()
-                }
-                if (ret instanceof Promise) {
-                  ret.then(handleRes.bind(null, id)).catch(handleError.bind(null, id))
-                } else if (ret instanceof Response) {
-                  handleRes(id, ret)
+        if (isAcmeChallengeRequest(req)) {
+          try {
+            handleAcmeChallenge(req).then(res => handleRes(id, res));
+          } catch (err) {
```

Comment:
This will only pick up if there's an error before the handler "becomes" a promise. I think you'd prefer a `.then().catch()` instead of the `try {} catch {}`

Reply:
Good catch